ducktape/command_line/main.py

```diff
@@ -56,6 +57,7 @@ def parse_args():
                              "This is primarily useful for test developers who want to interact with running " +
                              "services after a test has run. ")
     parser.add_argument("--version", action="store_true", help="display version")
+    parser.add_argument("--inject", action="store", help="Inject these arguments into the specified test. Use JSON to format.")
```

Comment:
Yeah, not requiring `@parametrize` or `@matrix` could be handy. I suspect it's not *that* big a deal to edit a method like that given that you're probably mucking with the code if you're running tests like that, but it's nice nonetheless.

I think my strongest negative reaction to the current approach is that even in the case where I just want to run an existing parameterization, I need to construct a carefully quoted JSON parameter.

That said, the two approaches aren't actually exclusive. We can easily make this version work, allowing you to pretty flexibly override test settings regardless of how the test is parameterized, and then later add test loader support for the other approach where you fully qualify the test name including parameters.